orchestrationSpecs/userConfigSchemas.txt

```diff
@@ -0,0 +1,59 @@
+HTTP_AUTH_BASIC: {
+  username: string
+  password: string
+}
+
+HTTP_AUTH_SIGV4: {
+  region:  string
+  service?: string | "es" // default
+}
+
+CLUSTER_CONFIG: {
+  name:            string
+  endpoint?:       string // optional for source and maybe metadata evaluate commands, but required for most others
+  allow_insecure?: bool
+  version?:        string // required in some contexts?
+  authConfig?:     HTTP_AUTH_BASIC | HTTP_AUTH_SIGV4
+}
+
+TARGET_CLUSTER_CONFIG: CLUSTER_CONFIG {
+  endpoint: string
+}
+
+SNAPSHOT_MIGRATION_CONFIG: {
+	indices: [string]
+	migrations: [{
+		metadata: {
+			mappings: {
+				properties: {...}
+			},
+			//
+			documentBackfillConfigs: {
+				indices: [...string],
+				config: {
+					batchSize: int,
+					initialReplicas: int
+				}
+			}
+		}
+	}]
+}
+
+REPLAYER_CONFIG: {
+	speedupFactor: float
```

Comment:
Why is the speedup factor a constant?  Should a user be able to dial this up and down during an execution of a workflow?

Reply:
Slightly further - why does a user set this value at all, shouldn't we auto-scale the replay speedup based on the target cluster performance just like RFS auto-scaling?